botocore/session.py

```diff
@@ -972,6 +892,52 @@ def lazy_register_component(self, name, no_arg_factory):
             pass
 
 
+class SessionVarsShim(collections.MutableMapping):
```

Comment:
I think it would be fine to just call this `SessionVarsMap`. The `Shim` part is a little vague for me.